docs/topic/whentouse.rst

```diff
@@ -0,0 +1,82 @@
+.. _topic/whentouse:
+
+===================================
+When to use The Littlest JupyterHub
+===================================
+
```

Comment:
GENERAL POINT, MAY INVALIDATE OTHER COMMENTS: We could probably remove a lot of this content, and boil it down to two main decision points:

1. Do you want your hub and all users to live on a single machine vs. putting them on a cluster of machines that are scaled up or down.
2. Do you need to use container technology of some kind. We should find some bells and whistles to motivate this.